Mathlib/CategoryTheory/IsConnected.lean

```diff
@@ -327,7 +327,7 @@ def Zigzag.setoid (J : Type u₂) [Category.{v₁} J] : Setoid J where
 /-- If there is a zigzag from `j₁` to `j₂`, then there is a zigzag from `F j₁` to
 `F j₂` as long as `F` is a functor.
 -/
-theorem zigzag_obj_of_zigzag (F : J ⥤ K) {j₁ j₂ : J} (h : Zigzag j₁ j₂) :
+theorem zigzag_obj_of_zigzag (F : J ⥤q K) {j₁ j₂ : J} (h : Zigzag j₁ j₂) :
```

Comment:
Do you really need the prefunctor version? I would suggest defining the prefunctor version as `zigzag_prefunctor_obj_of_zigzag`, while keeping the name `zigzag_obj_of_zigzag` for the functor version.

Reply:
I don't strictly need it because the prefunctor used in the construction is indeed a functor, but the proof is quite painful and the functoriality is not used at all.

Will revert as you proposed even though I think that `Zigzag`s are an inherently non-functorial construction since they're chains of existence proofs of morphisms without any coherence whatsoever.